### DISCUSSION TITLE
Adding term meta

### DISCUSSION THREAD

**Discussion by Vidyut** [publishpress/publishpress-taxonomies#1513] [Feature Requests]
Currently I add term meta using a function. And I use it to add semantic information like @type - person, organization, event, etc. As well as a url for sameAs, which I then use to add more detailed schema for the posts using the tags.

Since the plugin is already doing entity extraction, it may be possible to add term meta and autofill the type and sameAs from the information sent by the API - which probably gets wasted and discarded right now.

This would be a major enhancement. I can share the code I am using right now.

**Comment by stevejburge** [comment]
Thanks @Vidyut. I don't see the code yet

**Comment by Vidyut** [comment]
I am not so well versed with github pulls and my code uses ACF and Rank Math, but this is basically what I am doing. This code takes the custom fields with the @type and sameAs, as well as existing fields like description, etc and adds them to the Article schema for the post so that the default article schema is enhanced with semantic data like:
`"mentions: {"@type":"GovernmentOrganization", "name": "whatever ministry", "description": "blah blah", "url":"https://example.com/tag/simple-tag"; "sameAs":"wikipedia/wikidata/official url".... any additional specific schema...}`

`add_action( 'rank_math/vars/register_extra_replacements', function() {
   RankMath\Helper::register_var_replacement( 'vidmentions',array('VidyutMentions'),$args = array(
	        'name'    => esc_html__( 'Mentions Schema', 'rank-math' ),
			'desc'    => esc_html__( 'Adds tags with mentions schema', 'rank-math' )));	
} );
function VidyutMentions(){
	$terms =  get_the_tags($post_id);
	$copy = $terms;

	if ( !empty($terms) ){ 
  		foreach($terms as $term) {
  			//$mention = get_fields($term-); 
  			?>
  			{"@type" : "<?php if (!(get_field('type',$term))) echo 'Thing'; else the_field('type',$term); ?>",
  			"url" : "<?php echo ('https://aamjanata.com/tag/'. $term->slug .'/'); ?>",
  			"name" : "<?php echo $term->name ?>",
  			"description" : "<?php echo $term->description; ?>",
  			"sameAs" : "<?php the_field('url',$term); ?>"
  			<?php the_field('additional_schema',$term); ?>}
      <?php     if (next($copy )) {
        echo ','; }

		}
      $mentions = '['.$mention.']';
      return $mentions;
   }

}`

**Comment by Vidyut** [comment]
I am not a good coder. I was planning on using the Google NLP api to replace one of the APIs that no longer work and was planning to use the other data from the json response as well to auto fill description, type, sameas, etc when a tag is added to a post (or to update any missing fields in an existing tag that is also returned by the API). I am able to use the API as per their instructions, but hadn't got around to the code to replace in Simple Tags/now TaxoPress.

Obviously, if TaxoPress does this, the term fields will not be ACF ones and they will need to be added...

But I just love this plugin, so was planning to do this rather than code something different.

**Comment by Vidyut** [comment]
I will have some time next week and can try to make this fix work and share it back, but I suspect that it will need more coding/instructions to be used in the plugin for adding the google library, which needs composer... though there is also a version which can work with API keys or the SDK could be included or something I think. For my purposes, I am just hard coding everything, so it will probably not be a good idea to share back with git even if I figure out how to do that.

**Comment by stevejburge** [comment]
Thanks @Vidyut. We'll take a look at this and see if it's a good feature to add to TaxoPress. Thanks for the ideas
